### DISCUSSION TITLE
Errors after calling sodium_malloc

### DISCUSSION THREAD

**Discussion by Chewhern** [jedisct1/libsodium#1080] [Q&A] [answered]
I didn't encounter this problem when I test my binding at Windows 10 Home Single Language laptop, my laptop has 8 GB RAM, 2 standard cores. calling sodium_malloc through my binding generally has no problem.

But when I call this function on a ubuntu 18+ VPS with specification similar to my laptop only difference is the RAM is 2GB, I somehow encounter this problem which is coming from this line in sodium utils.
assert(_unprotected_ptr_from_user_ptr(user_ptr) == unprotected_ptr)

I noticed this problem when I was on my own SSH session, is there any system requirements(refer to the device or server specifications like RAM,Core etc) to use sodium_malloc? or is it only available to use specifically for Windows?

**Comment by jedisct1** [comment, marked as answer]
Did you initialize the library (`sodium_init()`) before using it?

**Reply by Chewhern** [reply]
It's strange.., but thanks, it kinda solved my problem.

When I tested it on my Windows device, it seems that it worked without calling the sodium_init()
